File: Project/spring-boot-login-mongodb-master/src/main/java/com/bezkoder/spring/menu/service/MealService.java
 package com.bezkoder.spring.menu.service;

 import com.bezkoder.spring.menu.MealModel.Meal;

 import java.util.List;



 public interface MealService {
     List<Meal> getAllMeals();
     Meal getMealById(String id);
     Meal createMeal(Meal meal);
     Meal updateMeal(String id, Meal meal);
     void deleteMeal(String id);
 }

File: Project/front/canteen/src/components/MealsService/MealService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/meals';

class MealService {
    getAllMeals() {
        return axios.get(API_URL);
    }

    createMeal(meal) {
        return axios.post(API_URL, meal);
    }

    updateMeal(id, meal) {
        return axios.put(`${API_URL}/${id}`, meal);
    }

    deleteMeal(id) {
        return axios.delete(`${API_URL}/${id}`);
    }
}

export default new MealService();
